clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -3148,6 +3200,70 @@ class SyclKernelBodyCreator : public SyclKernelFieldHandler {
     KernelHandlerClone->setInitStyle(VarDecl::CallInit);
   }
 
+  Expr *createArraySubscriptExpr(uint64_t Index, Expr *ArrayRef) {
+    QualType SizeT = SemaRef.getASTContext().getSizeType();
+
+    llvm::APInt IndexVal{
+        static_cast<unsigned>(SemaRef.getASTContext().getTypeSize(SizeT)),
+        Index, SizeT->isSignedIntegerType()};
+
+    auto IndexLiteral = IntegerLiteral::Create(
+        SemaRef.getASTContext(), IndexVal, SizeT, KernelCallerSrcLoc);
+
+    ExprResult IndexExpr = SemaRef.CreateBuiltinArraySubscriptExpr(
+        ArrayRef, KernelCallerSrcLoc, IndexLiteral, KernelCallerSrcLoc);
```

Comment:
The newlines between the declarations look a little odd, but that may be just me.

Reply:
I just moved existing code (L3329 -L3343 in old code) into this function and didn't modify it. I can remove it if you prefer.

Reply:
Okay, maybe wait for other code-reviewers, but it does look odd to me.

Reply:
>>he newlines between the declarations look a little odd, but that may be just me

Yes, Removing them would be nice. Thank you.